sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store.asciidoc

```diff
@@ -0,0 +1,602 @@
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 8 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension defines free functions for load/store operations within the
+group scope.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_GROUP_LOAD_STORE` to one of the values defined in the
+table below. Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Load API
+
+==== Single Value Overload
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load scalar
+template<typename Group, typename InputIteratorT,
+         typename OutputT, typename Properties = empty_properties_t>
+void group_load(Group g, InputIteratorT in_iter,
+                      OutputT& out, Properties = {});
+}
+----
+
+_Preconditions_: `in_iter` must be the same for all work-items
+in the group.
+
+._Constraints_:
+* Only available if `Group` is a work-group or sub-group.
+* `InputIteratorT` suppose to be a random access iterator.
+* Value type of `InputIteratorT` must be convertible to `OutputT`.
+* Value type of `InputIteratorT` and `OutputT` must be trivially copyable
+ and default constructible.
+
+_Effects_: Loads single element from `in_iter` to `out` by using the `g` group
+object to identify memory location as `in_iter` + `g.get_local_linear_id()`.
+
+`Properties` argument is reserved for future revisions of this extention and is
+ignored now. Default value is empty
+`sycl::ext::oneapi::experimental::empty_properties_t`.
+May be used in future for setting boundary values or limiting numbers of work
+items.
```

Comment:
Is this still true, or does the `contiguous` property make sense for this overload now?  If it does, I think you should include a similar note as the other overload:

```suggestion
Properties may provide a hint for optimization purposes
xref:optimization_hints[optimization_hints].
```